evm/utils/state_access_restriction.py

```diff
@@ -23,6 +23,40 @@ def is_accessible(address, slot_as_key, access_prefix_list):
     return False
 
 
+@to_set
+def remove_redundant_prefixes(prefix_list):
+    """Given a list of strings, this removes redundant strings that are covered
+    by other strings in the list. For example, given `["eth", "ethereum"]`,
+    this function will return just `["eth"]` as it's sufficient to match all
+    strings that are covered by both `eth` and `ethereum`.
+    """
+    class Node:
+        def __init__(self, word_end=False):
+            self.word_end = word_end
+            self.children = {}
+
+    root = Node(None)
+
+    for prefix in sorted(prefix_list, key=lambda prefix: len(prefix)):
+        cur = root
+        include = True
+
+        i = 0
+        while i < len(prefix):
```

Comment:
I think swapping this out for `for i in range(len(prefix))` will be more legible and won't require manually incrementing `i`.